include/deal.II/base/vectorization.h

```diff
@@ -5359,13 +5359,49 @@ namespace internal
   template <typename T>
   struct VectorizedArrayTrait
   {
-    using value_type = T;
+    using value_type                   = T;
+    static constexpr std::size_t width = 1;
+
+    static T &
+    get(T &value, unsigned int c)
+    {
+      AssertDimension(c, 0);
+      (void)c;
+
+      return value;
+    }
+
+    static const T &
+    get(const T &value, unsigned int c)
+    {
+      AssertDimension(c, 0);
+      (void)c;
+
+      return value;
+    }
```

Comment:
Such template specializations we also have somewhere else in the `matrix_free` folder; but I am not sure where. I think this or something similar at a central place would be great!

Reply:
I don't think we have precisely those. We have a number of related functionality for vectors/block vectors/vectors of vectors in `matrix_free/vector_access_internal.h`, and we have functions to determine the underlying `value_type` for the `FEEvaluationData` function, but I can't recall (and not find in a quick search) any related to making the interface between the vectorized array and regular numbers similar in terms of the access.